packages/flutter/lib/src/widgets/focus_scope.dart

```diff
@@ -314,45 +348,33 @@ class _FocusState extends State<Focus> {
   @override
   void didUpdateWidget(Focus oldWidget) {
     super.didUpdateWidget(oldWidget);
-    if (oldWidget.debugLabel != widget.debugLabel && _internalNode != null) {
-      _internalNode.debugLabel = widget.debugLabel;
-    }
-    if ((oldWidget.focusNode == widget.focusNode && oldWidget.onKey == widget.onKey)
-        || oldWidget.focusNode == null && widget.focusNode == null) {
-      // Either there aren't changes, or the _internalNode is already attached
-      // and being listened to.
+    assert(() {
```

Comment:
Is using an assert for only debug code the idiom flutter uses? I get how it works, but without the comment, I would not have known why this was in an assert.

Reply:
Yes, that's standard practice.